Implement 'Modify Guild Channel Positions'

gloss-water opened the pull request:
Currently the library implements the ['Create Channels' endpoint via POST to /guilds/{guild.id}/channels](https://discordapp.com/developers/docs/resources/guild#create-guild-channel), but doesn't implement the ['Modify Guild Channel Positions' endpoint](https://discordapp.com/developers/docs/resources/guild#modify-guild-channel-positions) at all.

This change adds that support as well as a method on Guild which will shuffle all existing channels using a new utility method for shuffling arrays.

iCrawl commented:
memes 👍 
![image](https://cloud.githubusercontent.com/assets/20760160/23100196/0495c03c-f67b-11e6-9afe-9cc83699e2b4.png)

Programmix commented:
This is a great PR but you mixed it with a meme... please separate the meme and the real PR thank you. 🙇

gloss-water commented:
Removed shuffleChannels method from the Guild object and replaced with a regular 'updateChannels' method.  Also removed the shuffleArray utility method since it would be unused.

GusCaplan commented:
shouldn't the method be named `updateChannelPositions` since that is all it changes

GusCaplan commented:
also `channelId` => `channelID` to keep with our standards

gloss-water commented:
I'll make those changes asap when I get home, thank you!

gloss-water commented:
Updated 'channelId' to 'channelID', I was also using 'guildId' instead of 'guildID' in RESTMethods so I updated that as well.

Gawdl3y commented on src/structures/Guild.js:
```diff
@@ -635,7 +635,21 @@ class Guild {
+
+  /**
+   * Updates this guild's channel positions as a batch.
+   * @param {Object[]} newChannelPositions Array of objects with an id and a position.
```
A typedef is needed, rather than `Object`.

gloss-water commented:
Should I do two typedefs, one for the array and one for the object inside?

Gawdl3y commented:
No, just one for the object.

gloss-water commented:
I updated the doc, please let me know if it needs more changes -- haven't worked with this style of documentation before. Thanks!

hydrabolt commented:
Looks good! 👍
